monitoring/mongodb/alerts.yaml

```
@@ -139,3 +139,68 @@ groups:
     annotations:
       description: 'MongoDb has low disk space'
       summary: 'MongoDb has low disk space'
+  
+  - alert: MongoDbPodRecovering
+    expr: |
+      mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*"} == 3
+    for: 1h
+    labels:
+      severity: warning
+    annotations:
+      description: "MongoDB pod `{{ $labels.pod }}` has been stuck in the 'RECOVERING' state for the past hour. The instance may not be able to join the replica set if the platform ingests a large number of operations during this time. This alert is expected if the 'Resync a Data Services MongoDB Member' procedure has recently been executed."
+      summary: MongoDB node recovering for too long
+
+  - alert: MongoDbPodRecovering
+    expr: |
+      mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*"} == 3
+    for: 1d
+    labels:
+      severity: critical
+    annotations:
+      description: "MongoDB pod `{{ $labels.pod }}` has been stuck in the 'RECOVERING' state for the past day. The instance may not be able to join the replica set if the platform ingests a large number of operations during this time."
+      summary: MongoDB node recovering for too long
+
+  - alert: MongoDbInvalidState
+    expr: |
+      mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", rs_state=~"6|8|10"} > 0
+    for: 5m
+    labels:
+      severity: critical
+    annotations:
+      description: "MongoDB pod `{{ $labels.pod }}` is repeatedly or continuously in the `{{ $labels.rs_state }}` state. Immediate attention is required. You may collect an sosreport for further analysis."
```

> Immediate attention is required
not useful if we don't provide a procedure

> You may collect an sosreport for further analysis
not really useful for the user, only for us if they open a ticket... but filling sosreport is part of the reporting process already

I think this should stay factual, and not try to deduce more than we can. If there is a will to provide more bits and pieces of the support documentation, should be a product-wide effort to update all alerts accross the product, with Techpub intervention.

```suggestion
      description: "MongoDB pod `{{ $labels.pod }}` is in an invalid state state (`{{ $labels.rs_state }}`)."
```

---

Given the alert rule, I think we should really say "repeatedly or continuously": we can be in the correct state when looking at the instance, yet the alert is still firing. This could be a symptom of a pod that is restarting in loop. Or we refine the alert rule to reflect this error if we are in this state continuously

---

the rule will be evaluated every 30s/1m, and it must match every time for 5 minutes : if we are unlucky enough that it matches every time prometheus evaluates it, then the user will also most likely see it....